lib/manager/poetry/artifacts.ts

```diff
@@ -107,7 +107,8 @@ function getSourceCredentialVars(
   const envVars: Record<string, string> = {};
 
   for (const source of poetrySources) {
-    const matchingHostRule = find({ url: source.url });
+    const matchingHostRule =
+      find({ hostType: 'pypi', url: source.url }) || find({ url: source.url });
```

Comment:
```suggestion
      find({ hostType: 'pypi', url: source.url }) ?? find({ url: source.url });
```

Will it make sense to use the datasource id for the hostType instead of a simple string here? So it will be future more safe.